telebot/async_telebot.py

```diff
@@ -183,6 +186,14 @@ def __init__(self, token: str, parse_mode: Optional[str]=None, offset: Optional[
         self.middlewares = []
 
         self._user = None # set during polling
+        self.bot_id = None
+
+        if token_check:
+            result = apihelper.get_me(token)
```

Comment:
@badiboy do you think this is a good idea

Reply:
So basically due to sync __init__, I had to call getMe from apihelper to obtain types.User due to token_check being true(new param)

Reply:
> So basically due to sync **init**, I had to call getMe from apihelper to obtain types.User due to token_check being true(new param)

Looks a bit ugly, but functional.